### PR TITLE
feat: article' CRUD

### PR DESCRIPTION
**Pull request by Baihhh**

### CODE REVIEW

File: cmd/gopcomm/community_yap.gox [hunk 1]
```diff
@@ -2,6 +2,8 @@ import (
 	"context"
 
 	"github.com/goplus/community/internal/core"
+	"github.com/goplus/community/markdown"
+	"time"
```
Comment (xushiwei) [inline cmd/gopcomm/community_yap.gox:6]: move to Go standard packages group.

File: internal/core/community.go [hunk 2]
```diff
@@ -19,15 +19,18 @@ package core
 import (
 	"context"
 	"database/sql"
-	"io"
+	"gocloud.dev/blob"
```
Comment (xushiwei) [inline internal/core/community.go:22]: move to user-defined packages group

File: internal/core/community.go [hunk 3]
```diff
@@ -80,55 +88,88 @@ func New(ctx context.Context, conf *Config) (ret *Community, err error) {
...
-	MarkEnd   = "eof"
-)
-
 // ListArticle lists articles from an position.
-func (p *Community) ListArticle(ctx context.Context, from string, limit int) (items []*ArticleEntry, next string, err error) {
-	if from == MarkBegin {
-		item := &ArticleEntry{
-			ID:    "123",
-			Title: "Title",
+func (p *Community) ListArticle(ctx context.Context, from int, limit int) (items []*ArticleEntry, next int, err error) {
```
Comment (xushiwei) [inline internal/core/community.go:155]: don't change func prototype. `from`, `next` should be `string` (not `int`) for implementation-independent.

File: internal/core/community.go [hunk 4]
```diff
@@ -80,55 +88,88 @@ func New(ctx context.Context, conf *Config) (ret *Community, err error) {
...
+		res, err := p.db.Exec(sqlStr, &article.Title, time.Now(), time.Now(), &article.UId, &article.Tags, &article.Cover, &article.Content)
+		if err != nil {
+			return "", err
+		}
+		idInt, err := res.LastInsertId()
+		return strconv.FormatInt(idInt, 10), nil
+	}
+	// edit article
+	// can uid edit the article
+	canEdit, err := p.CanEditable(ctx, uid, article.ID)
```
Comment (xushiwei) [inline internal/core/community.go:130]: No need to check access rights separately (should together with `update`).

File: internal/core/community.go (hunk 3 above)
Comment (xushiwei) [reply internal/core/community.go:155]: And all this api should provide unit tests.

**Review by IRONICBo — APPROVED** [review]
LGTM

**Pull request closed without merging**